server/sonar-server/src/main/java/org/sonar/server/platform/BackendCleanup.java

```diff
@@ -166,11 +174,49 @@ private static void truncateDefault(String tableName, Statement ddlStatement, Co
   }
 
   /**
+   * Admin membership on default organization must never be deleted
+   */
+  private static void truncateOrganizationsMembership(String tableName, Statement ddlStatement, Connection connection) throws SQLException {
+    Long adminId = selectAdminUserId(connection);
+    if (adminId == null) {
+      try (PreparedStatement preparedStatement = connection.prepareStatement("truncate table organization_members")) {
+        preparedStatement.execute();
+        // commit is useless on some databases
+        connection.commit();
+      }
+    } else {
+      try (PreparedStatement preparedStatement = connection.prepareStatement("delete from organization_members om where om.user_id <> ? ")) {
+        preparedStatement.setLong(1, adminId);
+        preparedStatement.execute();
+        // commit is useless on some databases
+        connection.commit();
+      }
+    }
+  }
+
+  @CheckForNull
+  private static Long selectAdminUserId(Connection connection) throws SQLException {
+    Statement stmt = connection.createStatement();
+    ResultSet rs = null;
+    try {
+      rs = stmt.executeQuery("select id from users ");
+      rs.next();
+      return rs.wasNull() ? null : rs.getLong(1);
+    } catch (SQLException ignored) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)